Add new vendor bitstream

kruczek8989 opened the pull request:
Add new vendor bitstream

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [ ] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

laf commented:
You've added probably 95% of the standard mibs LibreNMS ships, please remove them all. Any remaining ones need to be named correctly based on the MIB definition name, this is usually without the .mib or .txt

kruczek8989 commented:
> You've added probably 95% of the standard mibs LibreNMS ships, please remove them all. Any remaining ones need to be named correctly based on the MIB definition name, this is usually without the .mib or .txt

I put the MIB package I received from the device manufacturer (Bitstream) into the directory.
I'll take care of it tomorrow and organize this MIB folder ;)

PipoCanaja commented:
> > You've added probably 95% of the standard mibs LibreNMS ships, please remove them all. Any remaining ones need to be named correctly based on the MIB definition name, this is usually without the .mib or .txt
> 
> @laf I corrected and added the corrected MIB files

Seems that a lot of files are still standard MIB files that are already in the mibs dir of LibreNMS. Could you ensure that you don't add any duplicates ?

kruczek8989 commented:
> > > You've added probably 95% of the standard mibs LibreNMS ships, please remove them all. Any remaining ones need to be named correctly based on the MIB definition name, this is usually without the .mib or .txt
> 
> > 
> 
> > @laf I corrected and added the corrected MIB files
> 
> 
> 
> Seems that a lot of files are still standard MIB files that are already in the mibs dir of LibreNMS. Could you ensure that you don't add any duplicates ?

I compared the file list with the default MIB files and removed the duplicates.

PipoCanaja commented:
in the `bitsream` directory :
- BRIDGE-MIB -> Duplicate
- Dot3-OAM-MIB -> Duplicate
-ENTITY-MIB -> Duplicate
All the IEEEE and IANA are not from BTS either. 

In the `bitstream` directory, it is slightly better. But still : 
- SNMPv2-PDU
- SFLOW-MIB
etc

In fact, you could start by removing all MIB files with a name not starting with BTS, VTSS or H300, from bitstream directory and see how it goes. 
And completely remove the bitsream directory which looks like a leftover.

kruczek8989 commented:
> in the `bitsream` directory :
> 
> - BRIDGE-MIB -> Duplicate
> 
> - Dot3-OAM-MIB -> Duplicate
> 
> - ENTITY-MIB -> Duplicate
> 
> - etc
> 
> All the IEEEE and IANA are not from BTS either. 
> 
> 
> 
> In the `bitstream` directory, it is slightly better. But still : 
> 
> - SNMPv2-PDU
> 
> - SFLOW-MIB
> 
> - etc
> 
> 
> 
> In fact, you could start by removing all MIB files with a name not starting with BTS, VTSS or H300, from bitstream directory and see how it goes. 
> 
> And completely remove the bitsream directory which looks like a leftover.

But look at MIB Packard - improved. There were actually duplicate files in the first package.

laf commented:
Don't go off what's been done before.

For this PR to be reviewable you need to remove the standard MIBs we already have in the `mibs/` folder, rename any remaining ones to be the name as listed by DEFINTIONS within the file (most likely just without the .txt or .mib)

kruczek8989 commented:
> Don't go off what's been done before.
> 
> For this PR to be reviewable you need to remove the standard MIBs we already have in the `mibs/` folder, rename any remaining ones to be the name as listed by DEFINTIONS within the file (most likely just without the .txt or .mib)

But in last commit [MIB package - improved] I corrected the entire MIB package.

-rw-rw-r-- 1 librenms librenms   4577 wrz 19  2023 BTS-POE-MONITOR
-rw-rw-r-- 1 librenms librenms    720 gru 11 13:54 BTS-SMI
-rw-rw-r-- 1 librenms librenms  14066 sie 24  2022 BTS-TELEMETRY
-rw-rw-r-- 1 librenms librenms    722 gru 11 13:54 H300-SMI
-rw-rw-r-- 1 librenms librenms  12798 mar 26  2021 H300-TELEMETRY
-rw-rw-r-- 1 librenms librenms   4527 sie 28  2017 IANA-ADDRESS-FAMILY-NUMBERS-MIB
-rw-rw-r-- 1 librenms librenms  48032 sie 28  2017 IANA-IPPM-METRICS-REGISTRY-MIB
-rw-rw-r-- 1 librenms librenms  49400 gru 11 13:54 IEEE8021-MSTP-MIB
-rw-rw-r-- 1 librenms librenms  62588 sie 28  2017 IEEE8021-PAE-MIB
-rw-rw-r-- 1 librenms librenms 198524 sie 28  2017 IPATM-IPMC-MIB
-rw-rw-r-- 1 librenms librenms 182634 sie 28  2017 IPSEC-SPD-MIB
-rw-rw-r-- 1 librenms librenms  54450 sie 28  2017 MGMD-MIB
-rw-rw-r-- 1 librenms librenms  10372 gru 11 13:56 RADIUS-ACC-CLIENT-MIB
-rw-rw-r-- 1 librenms librenms  11610 gru 11 13:56 RADIUS-AUTH-CLIENT-MIB
-rw-rw-r-- 1 librenms librenms  25952 sie 28  2017 SFLOW-MIB
-rw-rw-r-- 1 librenms librenms   6494 sie 28  2017 SNMPv2-PDU
-rw-rw-r-- 1 librenms librenms  25340 sie 28  2017 VTSS-ACCESS-MANAGEMENT-MIB
-rw-rw-r-- 1 librenms librenms 110821 sie 28  2017 VTSS-ACL-MIB
-rw-rw-r-- 1 librenms librenms   9564 sie 28  2017 VTSS-AGGR-MIB
-rw-rw-r-- 1 librenms librenms  17859 sie 28  2017 VTSS-ARP-INSPECTION-MIB
-rw-rw-r-- 1 librenms librenms  33074 sie 28  2017 VTSS-AUTH-MIB
-rw-rw-r-- 1 librenms librenms  11122 sie 28  2017 VTSS-DAYLIGHT-SAVING-MIB
-rw-rw-r-- 1 librenms librenms  17745 sie 28  2017 VTSS-DDMI-MIB
-rw-rw-r-- 1 librenms librenms  21017 sie 28  2017 VTSS-DHCP6-CLIENT-MIB
-rw-rw-r-- 1 librenms librenms  10743 sie 28  2017 VTSS-DHCP-RELAY-MIB
-rw-rw-r-- 1 librenms librenms  65456 sie 28  2017 VTSS-DHCP-SERVER-MIB
-rw-rw-r-- 1 librenms librenms  22209 sie 28  2017 VTSS-DHCP-SNOOPING-MIB
-rw-rw-r-- 1 librenms librenms  14895 sie 28  2017 VTSS-DNS-MIB
-rw-rw-r-- 1 librenms librenms  11784 sie 28  2017 VTSS-EEE-MIB
-rw-rw-r-- 1 librenms librenms  21369 sie 28  2017 VTSS-EPS-MIB
-rw-rw-r-- 1 librenms librenms  39904 sie 28  2017 VTSS-ERPS-MIB
-rw-rw-r-- 1 librenms librenms 142842 sie 28  2017 VTSS-EVC-MIB
-rw-rw-r-- 1 librenms librenms   6803 sie 28  2017 VTSS-FAN-MIB
-rw-rw-r-- 1 librenms librenms  14962 sie 28  2017 VTSS-FIRMWARE-MIB
-rw-rw-r-- 1 librenms librenms   4931 sie 28  2017 VTSS-GVRP-MIB
-rw-rw-r-- 1 librenms librenms  15255 sie 28  2017 VTSS-HQOS-MIB
-rw-rw-r-- 1 librenms librenms   2661 sie 28  2017 VTSS-HTTPS-MIB
-rw-rw-r-- 1 librenms librenms  11781 sie 28  2017 VTSS-ICFG-MIB
-rw-rw-r-- 1 librenms librenms  49134 sie 28  2017 VTSS-IPMC-MVR-MIB
-rw-rw-r-- 1 librenms librenms  26101 sie 28  2017 VTSS-IPMC-PROFILE-MIB
-rw-rw-r-- 1 librenms librenms  78540 sie 28  2017 VTSS-IPMC-SNOOPING-MIB
-rw-rw-r-- 1 librenms librenms 187884 sie 28  2017 VTSS-IP-MIB
-rw-rw-r-- 1 librenms librenms  12094 sie 28  2017 VTSS-JSON-RPC-NOTIFICATION-MIB
-rw-rw-r-- 1 librenms librenms  17236 sie 28  2017 VTSS-LACP-MIB
-rw-rw-r-- 1 librenms librenms   5598 sie 28  2017 VTSS-LED-POWER-REDUCTION-MIB
-rw-rw-r-- 1 librenms librenms  83186 sie 28  2017 VTSS-LLDP-MIB
-rw-rw-r-- 1 librenms librenms   9020 sie 28  2017 VTSS-LOOP-PROTECTION-MIB
-rw-rw-r-- 1 librenms librenms  18381 sie 28  2017 VTSS-MAC-MIB
-rw-rw-r-- 1 librenms librenms 190233 sie 28  2017 VTSS-MEP-MIB
-rw-rw-r-- 1 librenms librenms 149379 sie 28  2017 VTSS-MPLS-MIB
-rw-rw-r-- 1 librenms librenms  43736 sie 28  2017 VTSS-MSTP-MIB
-rw-rw-r-- 1 librenms librenms  27287 sie 28  2017 VTSS-NAS-MIB
-rw-rw-r-- 1 librenms librenms   3642 sie 28  2017 VTSS-NTP-MIB
-rw-rw-r-- 1 librenms librenms   6950 gru 10  2021 VTSS-NTP-MONITOR
-rw-rw-r-- 1 librenms librenms  55169 sie 28  2017 VTSS-PERF-MONITOR-MIB
-rw-rw-r-- 1 librenms librenms  14414 sie 28  2017 VTSS-POE-MIB
-rw-rw-r-- 1 librenms librenms  50303 sie 28  2017 VTSS-PORT-MIB
-rw-rw-r-- 1 librenms librenms   9820 sie 28  2017 VTSS-PORT-POWER-SAVINGS-MIB
-rw-rw-r-- 1 librenms librenms   4137 sie 28  2017 VTSS-PRIVILEGE-MIB
-rw-rw-r-- 1 librenms librenms  13849 sie 28  2017 VTSS-PSEC-MIB
-rw-rw-r-- 1 librenms librenms  73888 sie 28  2017 VTSS-PTP-MIB
-rw-rw-r-- 1 librenms librenms   8604 sie 28  2017 VTSS-PVLAN-MIB
-rw-rw-r-- 1 librenms librenms 186269 sie 28  2017 VTSS-QOS-MIB
-rw-rw-r-- 1 librenms librenms   1030 sie 28  2017 VTSS-SMI
-rw-rw-r-- 1 librenms librenms  28713 sie 28  2017 VTSS-SNMP-MIB
-rw-rw-r-- 1 librenms librenms   3892 sie 28  2017 VTSS-SNTP-MIB
-rw-rw-r-- 1 librenms librenms   2075 sie 28  2017 VTSS-SSH-MIB
-rw-rw-r-- 1 librenms librenms  22699 sie 28  2017 VTSS-SYNCE-MIB
-rw-rw-r-- 1 librenms librenms   8501 sie 28  2017 VTSS-SYSLOG-MIB
-rw-rw-r-- 1 librenms librenms   8096 sie 28  2017 VTSS-SYSUTIL-MIB
-rw-rw-r-- 1 librenms librenms  16801 sie 28  2017 VTSS-TC
-rw-rw-r-- 1 librenms librenms   9194 sie 28  2017 VTSS-THERMAL-PROTECTION-MIB
-rw-rw-r-- 1 librenms librenms  21205 sie 28  2017 VTSS-TOPO-MIB
-rw-rw-r-- 1 librenms librenms  13220 sie 28  2017 VTSS-TT-LOOP-MIB
-rw-rw-r-- 1 librenms librenms   9418 sie 28  2017 VTSS-UDLD-MIB
-rw-rw-r-- 1 librenms librenms   3606 sie 28  2017 VTSS-UPNP-MIB
-rw-rw-r-- 1 librenms librenms   6210 sie 28  2017 VTSS-USERS-MIB
-rw-rw-r-- 1 librenms librenms  18289 sie 28  2017 VTSS-VCL-MIB
-rw-rw-r-- 1 librenms librenms  29292 sie 28  2017 VTSS-VLAN-MIB
-rw-rw-r-- 1 librenms librenms   9491 sie 28  2017 VTSS-VLAN-TRANSLATION-MIB
-rw-rw-r-- 1 librenms librenms  13696 sie 28  2017 VTSS-VOICE-VLAN-MIB

laf commented:
> But in last commit [MIB package - improved] I corrected the entire MIB package.

Maybe you haven't pushed the commit but I can assure you this PR contains far too many standard mibs.

PipoCanaja commented:
@kruczek8989 Cleaning done. Seems that we still have a lot of MIB files for these devices. And currently, in your discovery, you use none of them. Do you plan to extend the discovery ?

laf commented:
I'll tidy the MIBs up as we can't accept the pull request in it's current state.

However, one question about the OS, you have this as a Bitstream Hyperion 200 but what about the other hardware models in that range?

kruczek8989 commented:
> @kruczek8989 Cleaning done. Seems that we still have a lot of MIB files for these devices. And currently, in your discovery, you use none of them. Do you plan to extend the discovery ?


I received such a package from the manufacturer, but due to lack of time, I was not able to add all the things to use all the MIBs.



> I'll tidy the MIBs up as we can't accept the pull request in it's current state.
> 
> However, one question about the OS, you have this as a Bitstream Hyperion 200 but what about the other hardware models in that range?


I only have the Bitstream Hyperion 200 model

PipoCanaja commented on includes/definitions/hyperion.yaml:
```diff
@@ -0,0 +1,10 @@
+os: hyperion
+text: 'Bitstream Hyperion 200'
```
You can change this line to `text: 'Bistream Hyperion'`

PipoCanaja commented:
> > @kruczek8989 Cleaning done. Seems that we still have a lot of MIB files for these devices. And currently, in your discovery, you use none of them. Do you plan to extend the discovery ?
> 
> I received such a package from the manufacturer, but due to lack of time, I was not able to add all the things to use all the MIBs.

So either you want to only do a basic discovery (and we can probably wait a bit before adding 70+ files to LibreNMS) or you continue to improve the discovery and add sensors etc, in which case the necessary MIB files will be imported. But currently none of the MIB files are in use, at all. 

> > I'll tidy the MIBs up as we can't accept the pull request in it's current state.
> > However, one question about the OS, you have this as a Bitstream Hyperion 200 but what about the other hardware models in that range?
> I only have the Bitstream Hyperion 200 model

Even if we can't be sure, I would suggest to change the YAML os file and remove `200` from it, as we can suppose all the hyperion family should end up here.

laf reviewed (approved):
I've removed all the MIB files and added test data.

This should be good now once tests pass.

Thanks for contributing @kruczek8989, if you can provide a more complete walk I can add in additional support for you.

laf reviewed (dismissed):
Attempt #2

laf reviewed (approved):
Attempt #3

kruczek8989 commented:
> I've removed all the MIB files and added test data.
> 
> This should be good now once tests pass.
> 
> Thanks for contributing @kruczek8989, if you can provide a more complete walk I can add in additional support for you.


Thank you for your help and I'm sorry that I was absent for quite a long time, but at the moment I have to take time off from work due to family matters (my wife and I are expecting our third child :)).
I will update the current merge and possibly we will be able to further expand monitoring of this device model. Thank you again ;)
